server/task_check_scheduler.go

```diff
@@ -179,6 +179,74 @@ func (s *TaskCheckScheduler) Register(taskType string, executor TaskCheckExecuto
 
 // ScheduleCheckIfNeeded schedules a check if needed.
 func (s *TaskCheckScheduler) ScheduleCheckIfNeeded(ctx context.Context, task *api.Task, creatorID int, skipIfAlreadyTerminated bool) (*api.Task, error) {
+	// All task should pass timing task check.
+	// Since Time is an auto-increment value, a task would generate a timing task check every time before executing.
+	// However, this is an annoying behavior, and we adopt the following logic to ameliorate this problem.
```

Comment:
should be explicit about the exact logic instead of asking reader to understand the code. And also for this particular case, it's actually pretty tricky. The way the new logic is written is we would change the task run status, which is actually pretty surprising, given that method only expects to schedule a check (see the method comment)

also ameliorate is an uncommon word, we should choose simple word, e.g. ease